Add sensubility plugin

elfiesmelfie opened the pull request:
* Configure the exec plugin
* Enable the container healthcheck
* Render the collectd-sensubility.conf file

elfiesmelfie committed:
Add senubility plugin

* Configure the exec plugin
* Enable the container healthcheck
* Render the collectd-sensubility.conf file

elfiesmelfie committed:
Rename vars to tripleo_collectd_*

mrunge reviewed (commented):
In general, looks okay to me. Can you please fix the configure_healthcheck.yml linter error fixed, ideally cutting the line shorter than 79 chars, and also add a comment on the collectd_container_health_check_command (where the command is located).

mrunge reviewed (changes requested):
I already left a comment, describing the changes I would like to see.

csibbitt reviewed (commented):
I've given this a read-through and it LGTM

elfiesmelfie committed:
Update tasks/configure_healthcheck

Move openstack-healthcheck file into files/ and use copy
with src: <filename> instead of piping content into param

mrunge reviewed (approved):
In general, looks good to me.

paramite commented on tasks/configure_collectd.yml:
```diff
@@ -115,15 +116,88 @@
+      block:
+        - name: "Set sudoers path"
+          set_fact:
+            sensubility_sudoers_file: "{{ collectd_conf_output_dir }}/../sudoers.d/sensubility_collectd"
```
On puppet side the sudoers file name is composed from sensubility_ prefix and exec_user. I guess we don't need to hardcode this here, do we? IMO just sensubility is enough.

elfiesmelfie replied:
I was trying to keep the same files as puppet generates for the most part (exceptions are the exec.conf and python.conf) 
I'll take another look at how it's generated by puppet, but can also put in some values in defaults/main so these can be overwritten, if needed.

On the testing side, I have added some molecule tests in ``molecule/enable_sensubility/verify.yml``
I'd like some feedback on whether these are sufficient, and what additional steps you'd take to verify that things were working correctly. 
I can either include them in this role, or incorporate them into infrawatch/functional-tests, which are also run in downstream jenkins, for a scenarion that includes enabling sensubility.

On testing, this role can be installed on a tripleo node and tested with [1]
OR
you can run ``molecule test --destroy never -s enable_sensubility`` and  poke around.
If you want to check in the container, then run ``molecule login -s enable_sensubility`` to easily log in.

[1] https://review.opendev.org/c/openstack/tripleo-heat-templates/+/737100

paramite reviewed (approved):
One nit, but otherwise seems legit. I didn't test this though.

elfiesmelfie committed:
Merge branch 'master' into enable_sensubility

elfiesmelfie commented on tasks/configure_collectd.yml:
```diff
@@ -131,9 +129,82 @@
+        collectd_sensubility_log_level: "{{ tripleo_collectd_sensubility_log_level }}"
+        collectd_sensubility_transport: "{{ tripleo_collectd_sensubility_transport }}"
+        collectd_sensubility_connection: "{{ tripleo_collectd_sensubility_connection }}"
+        #collectd_sensubility_subscriptions:
```
I feel like this should be tripleo_collectd_sensubility_subscriptions, however, I have no idea where this value (or its equivalant (tripleo::profile::base::metrics::collectd::sensubility::subscriptions) comes from. 

It seems that in THT, all the MonitoringSubscription<Service> parameters are pushed to the outputs section of the heat templates as monitoring_subscriptions.

These are combined in the overcloud.yaml and used to set tripleo::profile::base::metrics::collectd::sensubility::subscriptions for puppet.

[1] https://github.com/openstack/tripleo-heat-templates/blob/master/overcloud.j2.yaml#L694

elfiesmelfie replied:
I need to update [2] to not set tripleo_collectd_sensubility_subscriptions, and then figure out how to pass the monitoring subscriptions properly.
I also need to update this to use tripleo_collectd_sensubility_subscriptions.
[2] https://review.opendev.org/c/openstack/tripleo-heat-templates/+/737100
